File: LeetCode/Easy/2053. Kth Distinct String in an Array/2024-01-28_Accepted.py
class Solution:
  def kthDistinct(self, arr: List[str], k: int) -> str:
    count = {}
    for s in arr:
      if s not in count:
        count[s] = 0
      count[s] += 1
    idx = 0
    for s in arr:
      if count[s] == 1:
        idx += 1
        if idx == k:
          return s
    return ''

File: LeetCode/Easy/2022. Convert 1D Array Into 2D Array/2024-01-28_Accepted.cpp
class Solution {
public:
  vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
    vector<vector<int>> result;
    if (m * n != original.size()) {
      return result;
    }
    int idx = 0;
    for (int i = 0; i < m; i++) {
      result.push_back(vector<int>(n));
      for (int j = 0; j < n; j++) {
        result[i][j] = original[idx++];
      }
    }
    return result;
  }
};
